Create dp-redis to allow health checking

morses-code opened the pull request:
Redis client with checker function for health checking.

Sanity check on code and make sure tests pass.

morses-code committed:
Create dp-redis to allow health checking

morses-code committed:
Delete .gitignore

morses-code committed:
Delete dp-redis.iml

morses-code committed:
Delete misc.xml

morses-code committed:
Delete modules.xml

morses-code committed:
Delete vcs.xml

morses-code committed:
Using miniredis to test Checker func

morses-code committed:
Removed test file

- Wasn't needed, was turning into an integration test

daiLlew commented on client.go:
```diff
@@ -0,0 +1,32 @@
+	Addr     string
+	Password string
+	DB       int
+	ttl      time.Duration
```
Should `ttl` be public? If not you won't be able to access it outside of this package.

morses-code replied:
You're right, miss type on my part.

daiLlew commented on client.go:
```diff
@@ -0,0 +1,32 @@
+package dp_redis
```
Double check with Eleanor on naming conventions for packages. Im not sure this follows Go idioms.

morses-code replied:
Again you're right, it doesn't. Package names shouldn't have underscores in them. Also don't think I should use `redis` as a package name considering I'm importing a redis package too...

daiLlew commented on client.go:
```diff
@@ -0,0 +1,32 @@
+}
+
+// Options - config options for the redis client
+type Options struct {
```
Would `Config` be a more appropriate name? It's more descriptive about what purpose it is serving.

morses-code replied:
I originally had this as `Config` only reason I changed it was to match the redis package wording as they call them `Options`.

gedge replied:
agreed - `Config` is more DP-style, too

daiLlew commented on client.go:
```diff
@@ -0,0 +1,32 @@
+type Options struct {
+	Addr     string
+	Password string
+	DB       int
```
What is `DB`?

morses-code replied:
Fair comment. It is database, `DB` is what it is called in the redis package I have imported too.

daiLlew commented on client.go:
```diff
@@ -0,0 +1,32 @@
+}
+
+// NewClient - returns new redis client with provided config options
+func NewClient(o Options) *Redis {
```
There should be some validation in here checking:

- The address is not empty
- Password is not empty
- ttl is a sensible value.

Personally I think t's better to return an error and fail to start up if its not right. Applying defaults is 'helpful' but ultimately masks the fact that something is not configured correctly.

morses-code replied:
That makes perfect sense. I'll be sure to add some validation in.

gedge commented on client.go:
```diff
@@ -0,0 +1,32 @@
+package dp_redis
+
+import (
+	"github.com/go-redis/redis"
```
I agree with Dave's comment about this file's package name - I would go with `package redis`
and then, for this line, use: 
```suggestion
	goredis "github.com/go-redis/redis"
```

to differentiate the imported package from this one

morses-code replied:
Didn't think of using an import alias.

gedge commented on client.go:
```diff
@@ -0,0 +1,32 @@
+)
+
+// Redis - structure for the redis client
+type Redis struct {
```
your editor (or, more generally, `go vet`) should warn you that this name is undesirable because users of this package would have to use something like `redis.Redis` which repeats, so consider:
```suggestion
type Client struct {
```

morses-code replied:
Makes sense, I think I've come across a few times where I've seen something similar. Renamed in next commit.

gedge commented on healthcheck.go:
```diff
@@ -0,0 +1,19 @@
+		return err
+	}
+	// Success
+	_ = state.Update(health.StatusOK, HealthyMessage, 0)
```
why ignore the error returned here (and four lines up)?

morses-code replied:
I found this odd too. I was using dp-s3 as a template. In s3 the line was simply: `state.Update(health.StatusOK, HealthyMessage, 0)` but looks like goland plugin for intellij changed it to what you see. Either way, I'm not sure what the purpose of it is.

gedge replied:
```suggestion
	return state.Update(health.StatusOK, HealthyMessage, 0)
```
is a minimal answer to this, pending further discussion (or maybe a ticket to address this as a bigger issue)

morses-code committed:
PR comments addressed

- Renamed dp_redis package to redis (dp_redis doesn't conform to go conventions)
- Redis struct renamed to Client
- Options struct renamed to Config
- ttl > TTL
- Added some validation against config values (address and password)

morses-code committed:
Updated as per PR comment

morses-code committed:
Update as per PR comment

morses-code committed:
Update to client and client test

- Added `Set` function for storing sessions in redis
- Added tests
- Created interface for client for mocking
- Mock file generated for client

morses-code committed:
Revert "Update to client and client test"

This reverts commit 64310e08bd682b587585aae6181a1861f216f066.
